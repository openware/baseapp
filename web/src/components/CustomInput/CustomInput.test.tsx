import * as React from 'react';
import { CustomInput, CustomInputProps } from './';

import { shallow } from 'enzyme';

const defaults: CustomInputProps = {
    type: '',
    label: '',
    defaultLabel: '',
    handleChangeInput: jest.fn(),
    inputValue: '',
    handleFocusInput: jest.fn(),
    classNameLabel: '',
    classNameInput: '',
    placeholder: '',
    autoFocus: false,
};

const setup = (props: Partial<CustomInputProps> = {}) =>
    shallow(<CustomInput {...{ ...defaults, ...props }} />);

describe('CustomInput component', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        const wrapper = setup();
        expect(wrapper).toBeDefined();
    });
});
