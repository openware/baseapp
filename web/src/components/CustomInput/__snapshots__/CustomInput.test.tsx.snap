// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput component should render 1`] = `
<Fragment>
  <div
    className="custom-input"
  >
    <label
      className=""
    />
    <InputGroup
      size="lg"
    >
      <FormControl
        autoFocus={false}
        onBlur={[MockFunction]}
        onChange={[Function]}
        onFocus={[MockFunction]}
        placeholder=""
        size="lg"
        type=""
        value=""
      />
    </InputGroup>
  </div>
</Fragment>
`;
